json_schema/system/links.json

```
@@ -128,33 +128,103 @@
           }
         }
       }
-    }
-  },
-  "properties": {
-    "describedBy": {
-      "description": "The URL reference to the schema.",
-      "type": "string",
-      "pattern": "^(http|https)://schema.(.*?)humancellatlas.org/system/(([0-9]{1,}.[0-9]{1,}.[0-9]{1,})|([a-zA-Z]*?))/links"
     },
-    "schema_version": {
-      "description": "The version number of the schema in major.minor.patch format.",
-      "type": "string",
-      "pattern": "^[0-9]{1,}.[0-9]{1,}.[0-9]{1,}$"
+    "supplementary_file": {
+    "type": "object",
+      "required": [
+        "file_id",
+        "file_type"
+      ],
+      "properties": {
+         "file_id": {
+            "description": "ID for this supplementary file entity",
+            "example": "111f222a-333b-444c-555d-eee63ddfccc7",
+            "type": "string",
+            "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$"
+          },
+         "file_type": {
+           "description" : "concrete type of this supplementary file entity",
+           "type": "string",
+           "enum": [
+             "supplementary_file"
+           ]
+         }
+      }
     },
-    "schema_type": {
-      "description": "The type of the metadata schema entity.",
-      "type": "string",
-      "enum": [
-        "links"
-      ]
+    "entity": {
+      "type": "object",
+      "required": [
+        "entity_id",
+        "entity_type"
+      ],
+      "properties": {
+         "entity_id": {
+            "description": "ID of this entity",
+            "example": "111f222a-333b-444c-555d-eee63ddfccc7",
+             "type": "string",
+             "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$"
+         },
+        "entity_type": {
+          "description": "The concrete type of this entity",
+          "type": "string"
+        }
+      }
     },
-    "links": {
-      "description": "An array of linking objects.",
-      "type": "array",
-      "items": {
-        "type": "object",
-        "$ref": "#/definitions/link"
+    "supplementary_file_link": {
```

Could shorten this to

```suggestion
    "supplementary_link": {
```

---

Wouldn't this suggest a link that is supplementary, and not a link to a supplementary file?

---

It would and I think that's ok. I can't imagine a scenario in which a link to a supplementary file is not supplementary itself. MW says "supplementary: available to supply something extra when needed" so I think supplementary-ness is transitive, so to speak.

However, this is a cosmetic issue for me. Either name works for me.

---

I'd also prefer to keep `_file_` as it could be confusing as we have the concept of 'supplementary links' to link to websites/github etc from the project

---

Agreed.